react/components/Resource.tsx

```diff
@@ -0,0 +1,237 @@
+import { ResourceCategory, License } from '../../typedql/schema'
+import { SelectOption } from '../../node/utils/types'
+import { PRODUCT_RESOURCES_PAGE } from '../common/globals'
+
+interface ResourceProps {
+  params: {
+    productId: number
+    resourceKey: string
+  }
+}
+
+function navigateToProductResourcesPage(context: any, props: ResourceProps) {
+  return () => {
+    context.navigate({
+      page: PRODUCT_RESOURCES_PAGE,
+      params: { productId: props.params.productId },
+    })
+  }
+}
+
+function Resource(props: ResourceProps) {
+  const defaultValue = {
+    id: null,
+    name: '',
+    resourceKey: '',
+    categoryId: '',
+    categoryName: '',
+    description: '',
+    licenses: [] as number[],
+    statements: {
+      actions: [''],
+      effect: 'allow',
+      resources: [
+        'vrn:{{service}}:{{region}}:{{account}}:{{workspace}}:{{path}}',
+      ],
+    },
+  }
+
+  const context = useRuntime()
+  const [formData, setFormData] = useState(defaultValue)
+  const [resourceCategories, setResourceCategories] = useState([])
+  const { loading, error, data } = useQuery(getResource, {
+    variables: {
+      productId: props.params.productId,
+      resourceKey: props.params.resourceKey,
+    },
+  })
+
+  useEffect(() => {
+    if (data && !loading && !error) {
+      setFormData(data.resource)
+      setResourceCategories(data.resourceCategories)
+    }
+  }, [loading, data, error])
+
+  const setFieldValue = useCallback(
+    (e: any) => {
+      setFormData({ ...formData, [e.target.name]: e.target.value })
+    },
+    [formData]
+  )
+
+  const defaultCategory = useCallback((categories, resource) => {
+    const match = categories.find(
+      (category: ResourceCategory) => category.id === resource.categoryId
+    )
+
+    if (match) {
+      return {
+        value: match,
+        label: match.name,
+      }
+    }
+
+    return null
+  }, [])
+
+  const defaultLicenses = useCallback((resource, licenses) => {
+    const match = licenses.filter((license: License) =>
+      resource.licenses.find(
+        (licenseId: number) => `${licenseId}` == license.id
+      )
+    )
+
+    if (match) {
+      return match.map((license: License) => ({
+        value: license.id,
+        label: license.name,
+      }))
+    }
+
+    return null
+  }, [])
+
+  if (loading) return <Spinner />
+  if (error) return <Alert type="error">Error: {error.message}</Alert>
+
+  return (
+    <div className="Resource">
+      <Layout
+        fullWidth
+        pageHeader={
+          <PageHeader
+            title="Edit resource"
+            linkLabel="Resources Manager"
+            onLinkClick={navigateToProductResourcesPage(context, props)}
+          />
+        }
+      >
+        <PageBlock>
+          <ToastProvider positioning="window">
+            <div className="flex justify-between mb5">
+              <div className="flex flex-column">
+                <b className="f5">General data</b>
```

Comment:
Nitpicking: just for future references, there's a `b` class! Also, do you really need the `f5`? This defines the font-size as 1rem which is supposed to be the default. If you do need it, try to use `t-body` instead. so this line could be replaced with:
```suggestion
                <span className="b t-body">General data</span>
```